src/dynamics/mpas/dyn_comp.F90

```diff
@@ -1028,6 +1045,69 @@ subroutine read_inidat(dyn_in)
 
    theta_m(:,1:nCellsSolve) = theta(:,1:nCellsSolve) * (1.0_r8 + Rv_over_Rd * tracers(ixqv,:,1:nCellsSolve))
 
+   ! if (.not. analytic_ic_active()) then  ! scale dry-air mass
+   !   surface_integral = cam_mpas_global_sum_real(areaCell(1:nCellsSolve))
+   !   write(iulog,*) subname//': Cell area test value = ', surface_integral
+   !   test_value = sqrt(surface_integral/(4.0_r8*pi))
+   !   write(iulog,*) subname//': earth radius from area = ', test_value
+   ! end if
+
+   call mpas_pool_get_config(domain_ptr % configs, 'config_scale_dry_air_mass', mpas_scale_dry_air_mass)
+
+   if (mpas_scale_dry_air_mass) then
+
+      allocate( p_top(nCellsSolve), preliminary_dry_surface_pressure(nCellsSolve), pm(plev) )
+      ! (1) calculate pressure at the lid                                                                                                         
```

Comment:
I would take this whole section and place it in a new routine.  SE and FV3 named similar routines set_dry_mass.  In addition, it needs to have log output similar to what SE and FV3 provide (ie. initial dry mass, scaled dry mass )

Reply:
Sounds good. I'll just create a new function in dyn_comp, `scale_dry_air_mass` and move this code there if that sounds good to you.

Reply:
Great. Feel free to use the FV3 dyn_comp.F90 as a template for types of logging messages.  They will help us validate the additional functionality.  Sounds like Peter is OK with adding these pressures in physconst.F90, maybe ps_dry_topo, ps_dry_notopo ! CESM reference dry mass pressure with/without topography.  I can change them during the PR to cam_development if needed.

Reply:
Moved to a new function. And added ps_dry_topo/notopo to physconst.F90.